Classes/Handler/ErrorHandler.php

```diff
@@ -22,153 +36,117 @@ class ErrorHandler
     protected $environment;
 
     /**
-     * @var \Raven_Client
+     * @param array $settings
      */
-    protected $client;
+    public function injectSettings(array $settings)
+    {
+        $this->dsn = $settings['dsn'] ?? '';
+        $this->environment = $settings['environment'] ?? '';
+    }
 
     /**
      * Initialize the raven client and fatal error handler (shutdown function)
      */
     public function initializeObject()
     {
-        $client = new \Raven_Client($this->dsn);
-        $errorHandler = new \Raven_ErrorHandler($client, true);
-        $errorHandler->registerShutdownFunction();
-        $this->client = $client;
+        \Sentry\init(['dsn' => $this->dsn]);
+
+        $options = Hub::getCurrent()->getClient()->getOptions();
+        $options->setEnvironment($this->environment);
+        $options->setRelease($this->getReleaseFromReleaseFile());
+        $options->setProjectRoot(FLOW_PATH_ROOT);
+        $options->setPrefixes([FLOW_PATH_ROOT]);
 
         $this->setTags();
-        $this->setEnvironment();
-        $this->setRelease();
-        $this->setAppPath();
-        $this->setPrefixes();
 
-        $this->emitSentryClientCreated($client);
+        $this->emitSentryClientCreated();
     }
 
     /**
-     * Explicitly handle an exception, should be called from an exception handler (in Flow or TypoScript)
+     * Explicitly handle an exception, should be called from an exception handler (in Flow or Fusion)
      *
      * @param object $exception The exception to capture
      * @param array $extraData Additional data passed to the Sentry sample
      */
-    public function handleException($exception, array $extraData = array())
+    public function handleException($exception, array $extraData = []): void
     {
-        if (!$this->client instanceof \Raven_Client) {
-            return;
-        }
 
```

Comment:
This check was necessary to stop Sentry from throwing "DSN not configured" exceptions in environments where Sentry is not actually used (e.g. in development). Newer versions of the Sentry client [seem to use a `NullTransport`](https://github.com/getsentry/sentry-php/blob/2.1.0/src/ClientBuilder.php#L315-L317) in that case, but I'd still vote to return early.

```suggestion
        if (empty($this->dsn)) {
            return;
        }

```